### DISCUSSION TITLE
Is there a group order?

### DISCUSSION THREAD

**Discussion by tonnico** [renovatebot/renovate#15796] [Request Help]
Hi,

is there an order for the grouping mechanism?

I'm using
```
        'group:allNonMajor',
        'group:aws-cdkMonorepo',
```
in my config.

Unfortunately the `aws-cdk` packages are note grouped together and are grouped in the `allNonMajor`-group.

**Comment by Gabriel-Ladzaretti** [comment]
You could try to [expand ](https://docs.renovatebot.com/presets-group/#groupaws-cdkmonorepo)`'group:allNonMajor',` and add `"excludePackagePatterns": ["^aws-cdk"],` to exclude packages starting with `aws-cdk` :
```
{
  "packageRules": [
    {
      "matchPackagePatterns": [
        "*"
      ],
      "excludePackagePatterns": ["^aws-cdk"],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch"
    }
  ]
}
```

**Reply by robosung** [reply]
I had the same question.  @Gabriel-Ladzaretti What is the general/predictable grouping behavior if there's overlap between several grouping declarations?

**Reply by rarkins** [reply]
Last matching rule wins

**Reply by robosung** [reply]
Thank you!

**Reply by quarhodron** [reply]
> Last matching rule wins

is is mentioned in the docs somewhere? I tried to find but I could not. Because even chat gpt seemingly does not know that and tells lie. I asked it:

>Let's assume that some package in renovate config matches two groupes simultaneously. In which group the package will be?
{
    "packageRules": [
        {
            "matchPackagePatterns": ["apollo-server"],
            "groupName": "Apollo server packages"
        },
        {
            "matchPackagePatterns": ["apollo-server", "metascraper"],
            "groupName": "Metascraper packages"
        }
    ]
}

it responded:
> In Renovate, a package can be part of multiple groups if it matches the criteria defined by multiple `packageRules`. The `groupName` is used to label the updates, but a single package can belong to multiple groups if it matches multiple rules.
In your provided configuration, if a package matches both `matchPackagePatterns` for "apollo-server", it will be included in both groups: "Apollo server packages" and "Metascraper packages".
Renovate does not prevent a package from being in more than one group, and each group would result in separate pull requests based on their configurations. These configurations allow the user to manage updates according to different strategies and preferences.

**Reply by rarkins** [reply]
It's described close to the top of the docs for `packageRules`.

![image](https://github.com/user-attachments/assets/99488b6e-6297-4240-83fe-9941ebb139e2)

You're welcome to submit a PR to improve this / make it more obvious if you'd like to help future users.
